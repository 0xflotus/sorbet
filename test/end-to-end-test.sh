#!/bin/bash
set -eux

# re-record with tools/scripts/update_exp_files.sh

diff -u test/end-to-end-test-output <(main/ruby-typer test/end-to-end-test-input.rb 2>&1 | sed -r -e 's,(rbi/stdlib.rbi:)[0-9]+,\1__LINE__,')
