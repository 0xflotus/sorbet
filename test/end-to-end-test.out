test/end-to-end-test-input.rb:4: Stubbing out unknown constant [5002]
     4 |    MissingConstant
            ^^^^^^^^^^^^^^^
test/end-to-end-test-input.rb:12: Argument arg0 does not match expected type String [7002]
    12 |    raise arg # raise is defined by stdlib
            ^^^^^^^^^
    rbi/stdlib.rbi:__LINE__: Method <overload N.1 : raise> has specified type of argument arg0 as String
    7054 |      arg0: String,
                ^^^^^
  Got Integer originating from:    test/end-to-end-test-input.rb:9: 
     9 |  sig(arg: Integer).returns(NilClass)
              ^^^^
test/end-to-end-test-input.rb:16: Argument arg does not match expected type Integer [7002]
    16 |    foo("foo")
            ^^^^^^^^^^
    test/end-to-end-test-input.rb:9: Method foo has specified type of argument arg as Integer
     9 |  sig(arg: Integer).returns(NilClass)
              ^^^^
  Got String originating from:    test/end-to-end-test-input.rb:16: 
    16 |    foo("foo")
                ^^^^^
